### ISSUE TITLE
List of Regions for the Plugin

### ISSUE THREAD

**Issue by veronicadumitriu** [oracle/oci-grafana-metrics#159]
Trying to create a new data source based on OCI metrics but my region is not listed in the Regions list (us-sanjose-1) - could you please add this region since it is a valid region and I cannot move the tenancy to another region once created.

thanks

**Comment by huskerdhf** [comment]
Agreed that the San Jose region should be added to the region dropdown on the datasource test page.

You can try the following workaround to see if you can move forward while the issue is being addressed:
1. When configuring the OCI metrics datasource, select another region such as us-phoenix-1.
2. The datasource test may fail but that doesn't prevent you from trying to use the OCI metrics datasource.
3. Create a dashboard and add a data panel, selecting the OCI metrics datasource as the datasource for the panel.
4. When you click on the region selection dropdown, the us-sanjose-1 region should appear in the list.

This should work since the regions list for the test data source page is statically defined while the regions list presented in the data panel editor is dynamically generated.

I just tried this with the 3.0.6 version of the OCI metrics plugin and the us-sanjose-1 region does appear in the regions list in the data panel editor.

**Comment by veronicadumitriu** [comment]
Hi - I have tried that and the list is empty. If i enter test manually, it will revert back to a nill selection.

**Comment by huskerdhf** [comment]
Hi, I think there must be broader issues occurring within the plugin. If you would share the Grafana log file for the timeframe where you tried to configure the OCI metrics datasource and then tried to use it in a data panel then we can take a look at it to see if we can identify the issue.

**Comment by veronicadumitriu** [comment]
Hi David,

I'm working with someone in the OCI team to see what is the source of the issue - it seems like the policy is not allowing the reading of the metrics - here is the log file from when trying to create the DS:

lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: ca92c4f64b1de3dc5542fb8f2d9d194a/6E19CBD47C711C545324DEDCAAB3E138/140867BC74091EE5ACB4E53C34691721" remote_addr=127.0.0.1
t=2022-12-07T22:54:33+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=893 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:34+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 1524d55384a2660c24b59264f347d777/0590B6F88C07E6CBB2A55DDC0A80666F/3DDBF7743F2D93B29166F511A5CE14AA" remote_addr=127.0.0.1
t=2022-12-07T22:54:34+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=39 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:35+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 75ea024d40a450e3e77c1dd5380ee6ce/90C6D30A14A2A139D40B75BB9D9EF4A6/0ADEAE703CAB7714E4995B1F8CAD8B14" remote_addr=127.0.0.1
t=2022-12-07T22:54:35+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=137 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:35+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: cbe25a99b370fb05913a158a84401761/C6EF3BE8B7B7A0E9398F8B03E07F6673/B10631DB5AB45EC555BF48383C1142B0" remote_addr=127.0.0.1
t=2022-12-07T22:54:35+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=11 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:36+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 7636820952e0945e9cb01f84261253e0/D8CE0549570A8091738EE3F3252DA369/E6A9BBC733C7BD8958006AB0A425593A" remote_addr=127.0.0.1
t=2022-12-07T22:54:36+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=16 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:36+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 73d36257116c1ed99f8f8df873167cfc/EBEDC958B7D1A55139BABD7D6BC4A580/71BEF80A98DDD7C17D94E35CFD27E1A5" remote_addr=127.0.0.1
t=2022-12-07T22:54:36+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=17 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:37+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 88281e7dae76d56bc8203d7311fc9fc7/0D9A6669FDE984D787AFE7D1F2F22924/09F31BCDC8B9DA8C841673BC285171A3" remote_addr=127.0.0.1
t=2022-12-07T22:54:37+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=17 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:37+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 6cd95a141a6bcee894289e2df8f1c0ce/05E965AC79D279CD4EB533CCD77C1AAC/FB1A47805DFB8F45EF8FCC8E2E28071D" remote_addr=127.0.0.1
t=2022-12-07T22:54:37+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=14 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:38+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 816c2ee3440b2336057ed97bfe5cf5ce/7392CCE37DCB3BA10ABD078D9D1BDA69/47D5525B0E6F972027E6D65C7B62F67E" remote_addr=127.0.0.1
t=2022-12-07T22:54:38+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=32 size=34 referer=http://localhost:3000/datasources/edit/2/
t=2022-12-07T22:54:39+0000 lvl=eror msg="Metric request error" logger=context userId=1 orgId=1 uname=admin error="rpc error: code = Unknown desc = Service error:NotAuthorizedOrNotFound. Authorization failed or requested resource not found.. http status code: 404. Opc request id: 574c23dd32f488a7dba4f3ca6db239b9/05BACE30648AA7A7FC2029AA1A0726A5/F5E56FC61D0A5EF602F4E5E3E3D5E037" remote_addr=127.0.0.1
t=2022-12-07T22:54:39+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/ds/query status=500 remote_addr=127.0.0.1 time_ms=19 size=34 referer=http://localhost:3000/datasources/edit/2/
